### PR TITLE
Restructure of the tabular submodule

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Part of #1406 

#### Any other comments?

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/suites.py [hunk 1]
```diff
@@ -30,5 +30,6 @@
     'train_test_leakage',
```
Comment (noamzbr) [inline deepchecks/suites.py:30]: Should remove single_dataset_integrity from here

File: deepchecks/tabular/checks/methodology/__init__.py [hunk 2]
```diff
@@ -35,3 +39,10 @@
     'ModelInferenceTime',
     'DatasetsSizeComparison',
 ]
+
+warnings.warn(
```
Comment (noamzbr) [inline deepchecks/tabular/checks/methodology/__init__.py:43]: How would we remember to remove this coming 0.8?

File: deepchecks/tabular/checks/methodology/__init__.py [hunk 3]
```diff
@@ -8,19 +8,23 @@
...
+        :mod:`deepchecks.tabular.checks.integrity` instead.
+
+"""
+import warnings
+
+from ..model_evaluation import BoostingOverfit, ModelInferenceTime, UnusedFeatures
+from ..train_test_validation import (DatasetsSizeComparison, DateTrainTestLeakageDuplicates,
+                                     DateTrainTestLeakageOverlap, IdentifierLeakage, IndexTrainTestLeakage,
+                                     SingleFeatureContributionTrainTest, TrainTestSamplesMix)
+from ..integrity import SingleFeatureContribution
```
Comment (noamzbr) [inline deepchecks/tabular/checks/methodology/__init__.py:26]: This wasn't included in the original methodology suite

**Review by noamzbr — COMMENTED** [review]
if the suite is data_integrity, shouldn't we call the module the same?

Also, need to do a general search for all uses of full_suite in examples and replace it with one of the individual suites.

File: deepchecks/tabular/checks/methodology/__init__.py (hunk 2 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/methodology/__init__.py:43]: We should have an issue for that on the 0.8 milestone

File: deepchecks/tabular/checks/methodology/__init__.py (hunk 3 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/methodology/__init__.py:26]: what do you mean? the checks here are the checks that were previously on the methodology folder

File: deepchecks/suites.py (hunk 1 above)
Comment (ItayGabbay) [reply deepchecks/suites.py:30]: I didn't remove it on purpose, in order to have the deprecation message

File: deepchecks/tabular/checks/integrity/__init__.py [hunk 4]
```diff
@@ -9,32 +9,36 @@
...
 from .mixed_nulls import MixedNulls
-from .new_label_train_test import NewLabelTrainTest
+# from .new_label_train_test import NewLabelTrainTest
 from .outlier_sample_detection import OutlierSampleDetection
+from .single_feature_contribution import SingleFeatureContribution
 from .special_chars import SpecialCharacters
 from .string_length_out_of_bounds import StringLengthOutOfBounds
 from .string_mismatch import StringMismatch
-from .string_mismatch_comparison import StringMismatchComparison
+# from .string_mismatch_comparison import StringMismatchComparison
```
Comment (JKL98ISR) [inline deepchecks/tabular/checks/integrity/__init__.py:26]: why so much commented code?

File: deepchecks/tabular/checks/methodology/__init__.py (hunk 3 above)
Comment (noamzbr) [reply deepchecks/tabular/checks/methodology/__init__.py:26]: Oh sorry, I was confused between "in the module" and "in the suite"

File: deepchecks/tabular/checks/integrity/__init__.py (hunk 4 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/integrity/__init__.py]: mistake. fixed

File: docs/source/checks/tabular/model_evaluation/plot_boosting_overfit.py [hunk 5]
```diff
@@ -86,7 +86,7 @@
 #%%
 # Run the check
 # ==============
-from deepchecks.tabular.checks.methodology.boosting_overfit import BoostingOverfit
+from deepchecks.tabular.checks.model_evaluation.boosting_overfit import BoostingOverfit
```
Comment (shir22) [inline docs/source/checks/tabular/model_evaluation/plot_boosting_overfit.py:89]: same - checks import location

File: docs/source/checks/tabular/model_evaluation/plot_model_inference_time.py [hunk 6]
```diff
@@ -11,7 +11,7 @@
 from sklearn.model_selection import train_test_split
 
 from deepchecks.tabular import Dataset
-from deepchecks.tabular.checks.methodology import ModelInferenceTime
+from deepchecks.tabular.checks.model_evaluation import ModelInferenceTime
```
Comment (shir22) [inline docs/source/checks/tabular/model_evaluation/plot_model_inference_time.py:14]: same in all...

**Review by nirhutnik — APPROVED** [review]

**Review by noamzbr — APPROVED** [review]

**Review by shir22 — APPROVED** [review]

**Pull request merged**